app/src/main/java/com/github/factotum_sdp/factotum/ui/directory/ContactCreation.kt

```diff
@@ -5,44 +5,115 @@ import android.view.LayoutInflater
+import androidx.navigation.findNavController
+import com.github.factotum_sdp.factotum.MainActivity
 import com.github.factotum_sdp.factotum.R
+import com.github.factotum_sdp.factotum.placeholder.Contact
 
 /**
  * A simple ContactCreation fragment.
  */
 class ContactCreation : Fragment() {
 
     // Should not stay like that and instead roles should use roles from future ENUM
-    val ROLES = listOf("Boss", "Courier", "Client")
+    private val roles = listOf("Boss", "Courier", "Client")
+    private var currentContact: Contact? = null
+    private val isUpdate: Boolean
+        get() = currentContact != null
+    private lateinit var viewModel : ContactsViewModel
 
-    override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
-    }
+    private lateinit var name: EditText
+    private lateinit var surname: EditText
+    private lateinit var address: EditText
+    private lateinit var phoneNumber: EditText
+    private lateinit var details: EditText
 
-    override fun onCreateView(
-        inflater: LayoutInflater, container: ViewGroup?,
-        savedInstanceState: Bundle?
-    ): View? {
-        // Inflate the layout for this fragment
+    private lateinit var spinner: Spinner
 
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        // Inflate the layout for this fragment
         return inflater.inflate(R.layout.fragment_contact_creation, container, false)
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        val spinner: Spinner = view.findViewById(R.id.roles_spinner)
-        //Initializes the spinner for the roles
+
+        spinner = view.findViewById(R.id.roles_spinner)
+        // Initializes the spinner for the roles
         ArrayAdapter(
             requireContext(),
             android.R.layout.simple_spinner_item,
-            ROLES
+            roles
         ).also { adapter ->
             // Specify the layout to use when the list of choices appears
             adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
             // Apply the adapter to the spinner
             spinner.adapter = adapter
         }
+
+        viewModel = ViewModelProvider(requireActivity())[ContactsViewModel::class.java]
+        viewModel.setDatabase(MainActivity.getDatabase())
+
+        if (arguments?.getInt("id") != null) {
+            currentContact = viewModel.contacts.value?.get(arguments?.getInt("id")!!)
+        }
+
+        setContactFields(view, currentContact)
+
+        val approveFormButton = view.findViewById<Button>(R.id.create_contact)
+
+        if (isUpdate) {
+            approveFormButton.text = getString(R.string.form_button_update)
+            approveFormButton.setOnClickListener {
+                viewModel.updateContact(
+                    Contact(
```

Comment:
You could use the apply function to set the values of the Contact object.

Reply:
True but contacts would then have to be a mutable type, and I don't really see the advantages. Since any contact we update we have to add to the database rewriting everything anyway.